odo/backends/blob.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import absolute_import, division, print_function
+
+from ..resource import resource
+from ..convert import convert, ooc_types
+from ..append import append
+from ..drop import drop
+import os
+import shutil
+import datashape
+from multipledispatch import MDNotImplementedError
+
+
+class File(object):
```

Comment:
FWIW the `py` library, from the same people who do `py.test` has objects like this that handle most of the operations that you're doing below. I'm not sure another dependency is in order, but it might be useful to see what they're doing over there: https://pylib.readthedocs.org/en/latest/